plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -547,13 +552,15 @@ private static Optional<Object> translateValue(Object trinoNativeValue, Type typ
         }
 
         if (type == TIMESTAMP_MILLIS) {
-            long millisUtc = (long) trinoNativeValue;
-            Instant instant = Instant.ofEpochMilli(millisUtc);
+            long epochMicros = (long) trinoNativeValue;
+            long epochSecond = floorDiv(epochMicros, MICROSECONDS_PER_SECOND);
+            int nanoFraction = floorMod(epochMicros, MICROSECONDS_PER_SECOND) * NANOSECONDS_PER_MICROSECOND;
+            Instant instant = Instant.ofEpochSecond(epochSecond, nanoFraction);
             return Optional.of(LocalDateTime.ofInstant(instant, UTC));
         }
 
         if (type == TIMESTAMP_TZ_MILLIS) {
-            long millisUtc = (long) trinoNativeValue;
+            long millisUtc = unpackMillisUtc((long) trinoNativeValue);
```

Comment:
In the testing - can we check if the filter reduces the number of rows processed by the system so that we could know if the filter is applied by the query engine or not.